### PR TITLE
Configurable drop target directory on windows

### PR DESCRIPTION
**Pull request by albertony**
Make drop target directory when ´--enable-drag-drop` is active configurable with new option `--drop-dir`. Default is Windows Desktop as before.

Related to https://github.com/debauchee/barrier/pull/969 and https://github.com/debauchee/barrier/issues/970, in the sense that the improve the ability to run as a "portable app" (e.g. copy-deploy with local configuration).

### CODE REVIEW

File: src/lib/platform/MSWindowsScreen.cpp [hunk 1]
```diff
@@ -1921,9 +1912,27 @@ std::string& MSWindowsScreen::getDraggingFilename()
     return m_draggingFilename;
 }
 
-const std::string& MSWindowsScreen::getDropTarget() const
+const std::string&
+MSWindowsScreen::getDropTarget()
```
Comment (p12tic) [inline src/lib/platform/MSWindowsScreen.cpp:1916]: I think such a case of cached data warrants `mutable` on `m_DropTargetPath`. This way the function can still be `const`.

**Review by p12tic — CHANGES_REQUESTED** [review]
Just a single comment, looks good otherwise. Thanks!

File: src/lib/platform/MSWindowsScreen.cpp (hunk 1 above)
Comment (albertony) [reply src/lib/platform/MSWindowsScreen.cpp:1916]: Sure. I didn't really like it that much myself, but saw it was done that way in DataDirectories class. I will make the getter const again and mark the variable as mutable then, (this is not a technique I've used a lot before either).

**Comment by albertony** [comment]
Pushed an update (mutable instead of const).

One thing I noticed: The OSXScreen has its own m_dropTarget member string, with a non-virtual getter to return it. Not sure why it is not virtual?

https://github.com/debauchee/barrier/blob/e66832c1d771c163f8afba7a957ac5aeb0dc98d9/src/lib/platform/OSXScreen.h#L102

**Comment by albertony** [comment]
Another small change: Increased log level of a few central messages. There are already quite some "noice" with info messages just when moving cursor between screens ("leaving screen", "entering screen", "switch from x to y" etc), so I though a larger operation such as transferring a file would then also qualify for a info message.

**Comment by p12tic** [comment]
> Pushed an update (mutable instead of const).
> 
> One thing I noticed: The OSXScreen has its own m_dropTarget member string, with a non-virtual getter to return it. Not sure why it is not virtual?
> 
> https://github.com/debauchee/barrier/blob/e66832c1d771c163f8afba7a957ac5aeb0dc98d9/src/lib/platform/OSXScreen.h#L102

`virtual` automatically carries over for functions that override the virtual function from the base class. So it can be omitted in theory. In practice this is error prone and we should mark the overriding functions with `override`.

**Review by p12tic — APPROVED** [review]
Looks good, thanks.

**Pull request merged**